backend/src/test/java/com/nikogrid/backend/services/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,346 @@
+
+    @InjectMocks
+    private AuthenticationService authenticationService;
+
+    private final String TEST_EMAIL = "test@example.com";
+    private final String TEST_PASSWORD = "password123";
+    private final String ENCODED_PASSWORD = "$2a$10$encodedPassword";
+    private final String JWT_TOKEN = "jwt.token.here";
+    private final RoleName TEST_ROLE = RoleName.OWNER;
+    private final UUID TEST_USER_ID = UUID.randomUUID();
+
+    private User testUser;
+    private UserDetailsImpl testUserDetails;
+    private AuthenticationDTO authenticationDTO;
+    private RegisterDTO registerDTO;
+    private Authentication mockAuthentication;
+
+    @BeforeEach
+    void setUp() {
+        testUser = createTestUser();
+        testUserDetails = new UserDetailsImpl(testUser);
+        authenticationDTO = new AuthenticationDTO(TEST_EMAIL, TEST_PASSWORD);
+        registerDTO = new RegisterDTO(TEST_EMAIL, TEST_PASSWORD, TEST_ROLE);
+        mockAuthentication = mock(Authentication.class);
+    }
+
+    // ==================== AUTHENTICATION TESTS ====================
+
+    @Test
+    @DisplayName("Should successfully authenticate user with valid credentials")
+    void authenticateUser_WithValidCredentials_ShouldReturnJwtToken() {
+        when(mockAuthentication.getPrincipal()).thenReturn(testUserDetails);
+        when(authenticationManager.authenticate(any(UsernamePasswordAuthenticationToken.class)))
+                .thenReturn(mockAuthentication);
+        when(jwtTokenService.generateToken(testUserDetails)).thenReturn(JWT_TOKEN);
+
+        RecoveryJwtTokenDTO result = authenticationService.authenticateUser(authenticationDTO);
+
+        assertNotNull(result);
+        assertEquals(JWT_TOKEN, result.token());
+
+        ArgumentCaptor<UsernamePasswordAuthenticationToken> tokenCaptor =
+                ArgumentCaptor.forClass(UsernamePasswordAuthenticationToken.class);
+        verify(authenticationManager, times(1)).authenticate(tokenCaptor.capture());
+
+        UsernamePasswordAuthenticationToken capturedToken = tokenCaptor.getValue();
+        assertEquals(TEST_EMAIL, capturedToken.getName());
+        assertEquals(TEST_PASSWORD, capturedToken.getCredentials());
+
+        verify(jwtTokenService, times(1)).generateToken(testUserDetails);
+        verifyNoInteractions(userRepository, securityConfiguration);
+    }
+
+    @Test
+    @DisplayName("Should throw exception when authentication fails")
+    void authenticateUser_WithInvalidCredentials_ShouldThrowBadCredentialsException() {
+        when(authenticationManager.authenticate(any(UsernamePasswordAuthenticationToken.class)))
+                .thenThrow(new BadCredentialsException("Invalid credentials"));
+
+        BadCredentialsException exception = assertThrows(
+                BadCredentialsException.class,
+                () -> authenticationService.authenticateUser(authenticationDTO)
+        );
+
+        assertEquals("Invalid credentials", exception.getMessage());
```

Comment:
You are validating that the message is the one you set in the mock??? This isn't testing anything.